### PR TITLE
489 - Streamed comments arrive twice

### PR DESCRIPTION
**Pull request by giladt**
## What was done
- [x] Fixed an issue causing comments that arrive from the stream, to appear multiple times in some cases.
- [x] Fixed an error raised when `refComments` do not exist when no errors on the list.
- [x] Removed `comments` object from service. Comments are handled solely in `discussionThread`.

### CODE REVIEW

File: src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts [hunk 1]
```diff
@@ -312,9 +343,15 @@ export class DiscussionThread {
     if (this.isLoading[`isDeleting ${_id}`]) return;
     this.isLoading[`isDeleting ${_id}`] = true;
     try {
-      this.threadComments = await this.discussionsService.deleteComment(this.discussionId, _id);
+      const swrComments = [...this.threadComments];
```
Comment (hiaux0) [inline src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts:346]: What does swr mean?

File: src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts [hunk 2]
```diff
@@ -272,6 +276,27 @@ export class DiscussionThread {
...
+          _id: newComment._id,
+          text: newComment.text,
+          author: newComment.author,
+          authorENS: newComment.authorENS,
+          metadata: newComment.metadata,
+          replyTo: newComment.replyTo,
+          upvotes: newComment.upvotes,
+          downvotes: newComment.downvotes,
+          createdOn: newComment.createdOn,
+        });
```
Comment (hiaux0) [inline src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts:291]: possible?:

1. 
```suggestion
        this.threadComments.push(newComment);
```

or

2.
```suggestion
        this.threadComments.push({ ...newComment });
```

File: src/dealDashboard/discussionsService.ts [hunk 3]
```diff
@@ -297,37 +291,41 @@ export class DiscussionsService {
...
-      this.comments = await this.convo.comments.query({
+      const comments = (await this.convo.comments.query({
         threadId: `${discussionId}:${this.getNetworkId(process.env.NETWORK as AllowedNetworks)}`,
-      });
-      if (!this.comments || this.comments.length === undefined) return null;
-
-      this.comments = await this.comments.filter((comment: any) => {
-        return !(
-          (comment.metadata.isPrivate === "true") &&
+      })).filter((comment: any) => !(
```
Comment (hiaux0) [inline src/dealDashboard/discussionsService.ts:296]: Would be nice to have a type for the convo api response

File: package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
     "@makerdao/multicall": "^0.12.0",
     "@metamask/detect-provider": "^1.2.0",
     "@stackoverflow/stacks": "^0.74.0",
-    "@theconvospace/sdk": "^0.3.29",
+    "@theconvospace/sdk": "^0.3.37",
```
Comment (hiaux0) [inline package.json:43]: Updating a minor version shouldn't have such a large amount of diff (52k) in `package-lock.json` :thinking: 

I tried it locally with much less changes
image

**Review by hiaux0 — CHANGES_REQUESTED** [review]
Change request due to package-lock.json file, and its large amount of diffs

Other comments are optional.

(only code was reviewed, overall good)

File: src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts (hunk 1 above)
Comment (giladt) [reply src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts:346]: stale while revalidates:
https://swr.vercel.app/

File: package.json (hunk 4 above)
Comment (giladt) [reply package.json]: Reverted that change for now, as it not really nesessary for this PR.

**Review by dkent600 — APPROVED** [review]

**Pull request merged**